pkg/sql/distsqlrun/tablereader.go

```diff
@@ -245,9 +249,18 @@ func (tr *tableReader) Start(ctx context.Context) context.Context {
 
 	// This call doesn't do much; the real "starting" is below.
 	tr.input.Start(fetcherCtx)
+
+	limitBatches := true
+	// We turn off limited batches if we know we have no limit and if all of the
+	// tableReader spans are point spans. This enables distsender parallelism -
+	// if limitBatches is true, distsender does *not* parallelize multi-range
+	// scan requests.
+	if tr.allPointSpans && tr.limitHint == 0 {
```

Comment:
Maybe only do this if we don't have a ton of spans?

Reply:
Seems like that would be taking us even deeper into random heuristic territory. What would you propose as a cutoff? I think we should just hand it off to distSender, which will naturally limit parallelism for us.

Reply:
Not really worried about parallelism, but about having to hold all results in memory at once.. I'd use the default batch size as a cutoff.

Reply:
But then beyond that cutoff, you'd have a horrible performance degradation: you'd revert to fully serialized execution. Well, I'll add it as protection I suppose.

Reply:
Actually I still think this doesn't make too much sense. What "default batch size" are you talking about, btw?

Reply:
`kvBatchSize` in kvfetcher.go. I don't feel strongly about this (you can make the argument that the size of the KVs is proportional to the size of the spans which must be in memory regardless), your call.